clients/callback_client.go

```diff
@@ -86,21 +90,54 @@ func (c CallbackClient) SendTranscodeStatusError(callbackURL, errorMsg string) e
 	return c.DoWithRetries(r)
 }
 
+// Calculate the overall completion ratio based on the completion ratio of the current stage.
+// The weighting will need to be tweaked as we understand better the relative time spent in the
+// segmenting vs. transcoding stages.
+func overallCompletionRatio(status TranscodeStatus, currentStageCompletionRatio float64) float64 {
+	// Sanity check the inputs are within the 0-1 bounds
+	if currentStageCompletionRatio < 0 {
+		currentStageCompletionRatio = 0
+	}
+	if currentStageCompletionRatio > 1 {
+		currentStageCompletionRatio = 1
+	}
+
+	// Define the "base" numbers - e.g the overall ratio we start each stage at
+	var TranscodeStatusPreparingBase float64 = 0.0
+	var TranscodeStatusPreparingTranscodingBase float64 = 0.4
```

Comment:
Should it be only "Transcoding"?
```suggestion
	var TranscodeStatusTranscodingBase float64 = 0.4
```

Reply:
The "Base" was trying to signify that this is the lower limit / floor / whatever, open to suggestions for better names though. Maybe "Floor"?

Reply:
Just realised what you were actually pointing out 👓